drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/editor/ScenarioSimulationEditorKogitoWrapper.java

```diff
@@ -156,14 +171,20 @@ public Promise setContent(String fullPath, String value) {
             }
             final Path path = PathFactory.newPath(finalName, pathString);
 
-            if (value == null || value.isEmpty()) {
-                newFile(path);
+            /* If given content is null, a new file has to be created. */
+            /* Otherwise, the content is un-marshalled and shown in the editor */
+            if (content == null || content.isEmpty()) {
+                showScenarioSimulationCreationPopup(path);
```

Comment:
I checked that manually and it was not working
If I pass malformed scesim then it just shows the exception 
![image](https://user-images.githubusercontent.com/21653197/78335309-abb7a800-758d-11ea-8422-810d44d91fda.png)

Are you sure that it does not propagates to the Exception part skipping   unmarshallContent(content)

Reply:
@dupliaka In my opinion, an exception is thrown inside `unmarshallContent(content)` and is caught in the catch of line 183. Therefore, the behavior is correct in my opinion. Maybe we can improve the message

Reply:
Ah, ok. Sorry.